pytorch_lightning/trainer/data_loading.py

```diff
@@ -79,27 +78,19 @@ def init_train_dataloader(self, model):
             self.val_check_batch = int(self.num_training_batches * self.val_check_interval)
             self.val_check_batch = max(1, self.val_check_batch)
 
-        on_ddp = self.use_ddp or self.use_ddp2
-        if on_ddp and not isinstance(self.get_train_dataloader().sampler, DistributedSampler):
-            msg = """
-            You're using multiple gpus and multiple nodes without using a DistributedSampler
-            to assign a subset of your data to each process. To silence this warning, pass a
-            DistributedSampler to your DataLoader.
-
-            ie: this:
-            dataset = myDataset()
-            dataloader = Dataloader(dataset)
-
-            becomes:
-            dataset = myDataset()
-            dist_sampler = torch.utils.data.distributed.DistributedSampler(dataset)
-            dataloader = Dataloader(dataset, sampler=dist_sampler)
+        self._ddp_sampler_check(self.get_train_dataloader(), 'train')
 
-            If you want each process to load the full dataset, ignore this warning.
-            """
-            if msg not in self.shown_warnings and self.proc_rank == 0:
-                self.shown_warnings.add(msg)
-                warnings.warn(msg)
+        # support IterableDataset for train data
+        self.is_iterable_train_dataloader = (
+            isinstance(self.get_train_dataloader(), DataLoader) and
+            EXIST_ITER_DATASET and isinstance(self.get_train_dataloader().dataset, IterableDataset))
```

Comment:
the name `EXIST_ITER_DATASET` is a bit misleading... shall it ba also available?

Reply:
This change is mostly about checking if we are getting a pytorch dataloader before checking its member. 
I suppose the naming could be the same as others. ex. `ITER_DATASET_AVAILABLE`?

Reply:
yeah, `ITER_DATASET_AVAILABLE` sounds better...